prow/config/config.go

```diff
@@ -216,14 +216,39 @@ type InRepoConfig struct {
 	AllowedClusters map[string][]string `json:"allowed_clusters,omitempty"`
 }
 
+func trimRepoPrefix(repo string) string {
+	repo = strings.TrimPrefix(repo, "http://")
+	repo = strings.TrimPrefix(repo, "https://")
+	return repo
+}
+
+func SplitRepoName(repo string) (string, string, error) {
+	// Normalize repo name to remove http:// or https://, this is the case for some
+	// of the gerrit instances.
+	repo = trimRepoPrefix(repo)
+	s := strings.SplitN(repo, "/", 2)
+	if len(s) != 2 {
+		return "", "", fmt.Errorf("repo %s cannot be split into org/repo", repo)
+	}
+	return s[0], s[1], nil
+}
+
 // InRepoConfigEnabled returns whether InRepoConfig is enabled for a given repository.
+// Assumes that config will not include http:// or https://
 func (c *Config) InRepoConfigEnabled(identifier string) bool {
-	if c.InRepoConfig.Enabled[identifier] != nil {
-		return *c.InRepoConfig.Enabled[identifier]
+	// SplitRepoName trims http:// and https:// from beginning of org name
+	split := false
+	org, _, err := SplitRepoName(identifier)
+	if err == nil {
+		split = true
```

Comment:
I wonder if we can do away with this `split` boolean as it makes the code a bit harder to read, as we have to keep track of that state to reason about the rest of the code.

But also, isn't this a bug? If we return a non-nil error, then `org` is the empty string, yet we still proceed with the rest of this logic.

Reply:
Theoretically we do not need the split boolean and we can just use err==nil instead of split==true, but I thought that would be harder to understand.

Reply:
> Below we do not use org if split is false.

Shouldn't it be, "We do not use org if it is empty"?

In general I am suspicious of variables that have an exact 1:1 mapping (here, if split == true, then err == nil, if split == false then err == non-nil; and these mappings will always be the same).

Maybe I'm over-thinking this, curious to see what others think. @cjwagner @chaodaiG

Reply:
Oh I actually totally agree. I was a little blinded by the error, but just checking if org != "" is much easier.